src/chains/evm/hub/utils/message.ts

```diff
@@ -108,27 +111,29 @@ export async function getHubReverseMessageExtraArgsAndAdapterFees(
   message: MessageReceived,
   extraArgsParams: ReverseMessageExtraArgsParams,
   payload: Payload,
+  adapterId: AdapterType,
 ): Promise<{
   adapterFees: bigint;
   extraArgs: ReverseMessageExtraArgs;
 }> {
   const { action, data } = payload;
   const payloadData = decodeMessagePayloadData(action as ReversibleHubAction, data);
+  const folksTokenId = getFolksTokenIdFromPool(payloadData.poolId);
 
-  const returnAdapterId = extraArgsParams?.returnAdapterId ?? message.returnAdapterId;
+  const returnAdapterId = extraArgsParams?.returnAdapterId ?? adapterId;
```

Comment:
Keep it as `const returnAdapterId = extraArgsParams?.returnAdapterId ?? message.returnAdapterId;`

Reply:
in that case the front end have to override it and not leave the default one (the default one is the hub adapter id)

Reply:
The frontend should set the return adapter id in operations like deposit to consider the possibility of having to reverse. We can add a check in the SDK too that the `returnAdapterId` used in the `prepare.deposit` supports message